12-offer-encoding.md

```diff
@@ -0,0 +1,769 @@
+      program
+  - if it is connected only by private channels:
+    - MUST include a `blinded_path` containing one or more paths to the node.
+  - otherwise:
+    - MAY include `blinded_path`.
+  - if it includes `blinded_path`:
+    - MUST specify `path` in order of most-preferred to least-preferred if
+      it has a preference.
+    - MUST include `blindedpay` with exactly one `payinfo` for
+      each `onionmsg_path` in `blinded_path`, in order.
+    - if it includes `blinded_capacities`:
+      - MUST include exactly one `incoming_msat` (in millisatoshis) per `path`, reflecting the expected maximum amount that can be sent through the path.
+    - SHOULD ignore any payment which does not use one of the paths.
+  - otherwise:
+    - MUST NOT include `blinded_payinfo`.
+  - MUST specify `amount`.`msat` in multiples of the minimum lightning-payable unit
+    (e.g. milli-satoshis for bitcoin) for `chain` (or for bitcoin, if there is no `chain`).
+  - if responding to an `invoice_request`:
+    - if for the same `offer_id` and `payer_key` as a previous `invoice_request`:
+      - MAY simply reuse the previous invoice.
+    - otherwise:
+      - MUST NOT reuse a previous invoice.
+    - MUST set `node_id` the same as the offer.
+    - MUST set (or not set) `quantity` exactly as the invoice_request did.
+    - MUST set `payer_key` exactly as the invoice_request did.
+    - MUST set (or not set) `payer_info` exactly as the invoice_request did.
+    - MUST set (or not set) `payer_note` exactly as the invoice_request did,
+      or MUST not set it.
+    - MUST set (or not set) `replace_invoice` exactly as the invoice_request did.
+    - MUST begin `description` with the `description` from the offer.
+    - MAY append additional information to `description` (e.g. " +shipping").
+    - if it does not set `amount` to the *base invoice amount* calculated from the invoice_request:
+       - MUST append the reason to `description` (e.g. " 5% bulk discount").
+    - MUST set (or not set) `issuer` exactly as the offer did.
+    - MUST NOT set `refund_for`
+    - MUST NOT set `refund_signature`
+  - otherwise (responding to a `send_invoice` offer):
+    - MUST set `node_id` to the id of the node to send payment to.
+    - MUST set `description` the same as the offer.
+    - if the offer had a `quantity_min` or `quantity_max` field:
+      - MUST set `quantity`
+      - MUST set it within that (inclusive) range.
+    - otherwise:
+      - MUST NOT set `quantity`
+    - MUST set `payer_key` to the `node_id` of the offer.
+    - MUST NOT set `payer_info`.
+    - MUST set (or not set) `refund_for` exactly as the offer did.
+    - if it sets `refund_for`:
+      - MUST set `refund_signature` to the signature of the
+        `refunded_payment_hash` using prefix `refund_signature` and the `payer_key` from the to-be-refunded invoice.
+    - otherwise:
+      - MUST NOT set `refund_signature`
+
+A reader of an invoice:
```

Comment:
Not trivial at all!  I filled this out in various places; basically if it's in response to an invoice_request it has to copy the field, if it's for a `send_invoice` it has to be one of the chains allowed by that offer, otherwise it simply has to be a supported chain.